data.md

```diff
@@ -245,10 +245,13 @@ Bitwise Operators
                  | twos ( Int , Int )       [function]
                  | bswap ( Int , Int )      [function]
  // --------------------------------------------------
-    rule signextend(N, W) => twos(N +Int 1, W) -Int (1 <<Byte (N +Int 1))  requires         word2Bool(bit((8 *Int (N +Int 1) -Int 1), twos(N +Int 1, W)))
-    rule signextend(N, W) => twos(N +Int 1, W)                             requires notBool word2Bool(bit((8 *Int (N +Int 1) -Int 1), twos(N +Int 1, W)))
+    rule signextend(N, W) => twos(N, W) -Int (1 <<Byte N)  requires N>Int 0 andBool        word2Bool(bit((8 *Int N -Int 1), twos(N, W)))
+    rule signextend(N, W) => twos(N, W)                    requires N>Int 0 andBool notBool word2Bool(bit((8 *Int N -Int 1), twos(N, W)))
+    rule signextend(0, _) => 0
 
     rule twos(N, W) => W modInt (1 <<Byte N)
+      requires N >Int 0
+    rule twos(0, _) => 0
```

Comment:
would our modInt break if the case above handled `N >=Int 0`?

Reply:
<<Byte would break because it relies on <<Int which assumes that the right operand is greater than zero.

Reply:
Ok